### ISSUE TITLE
foldl vs. List.fold_left

### ISSUE THREAD

**Issue by jdeisenberg** [darklang/tablecloth#18]
These produce different output:
```
let _ = Js.log(foldl ~f:(-) ~init:0 [1;2;3]) (* output is 2 *)
let _ = Js.log(List.fold_left (-) 0 [1;2;3]) (* output is -6 *)
```
It looks like the reducer function for`List.fold_left` takes the accumulator and list item as its parameters (in that order); `foldl` takes the list item and accumulator.

Possible performance issue: the code for `foldl` in `bs/src/tablecloth.ml` uses `List.fold_right`, which is not tail recursive, according to http://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html

**Comment by pbiggar** [comment]
Thanks for the report! Definitely something to fix.

**Comment by j-m-hoffmann** [comment]
Allready fixed that, but it is in my Add Array module branch

**Comment by j-m-hoffmann** [comment]
spoke a little too soon but i am on it

**Comment by j-m-hoffmann** [comment]
fixed in PR #26

**Comment by j-m-hoffmann** [comment]
check out PR #58
